src/transaction.ts

```diff
@@ -458,3 +492,215 @@ export class TransactionHash extends Hash {
+   */
+  public signature: Uint8Array;
+
+  /**
+   * The signature of the guardian.
+   */
+  public guardianSignature: Uint8Array;
+
+  /**
+   * Creates a new Transaction object.
+   */
+    public constructor({
+      sender,
+      receiver,
+      gasLimit,
+      chainID,
+      nonce,
+      value,
+      senderUsername,
+      receiverUsername,
+      gasPrice,
+      data,
+      version,
+      options,
+      guardian,
+    }: {
+      nonce?: BigNumber.Value;
+      value?: BigNumber.Value;
+      sender: string;
+      receiver: string;
+      senderUsername?: string;
+      receiverUsername?: string;
+      gasPrice?: BigNumber.Value;
+      gasLimit: BigNumber.Value;
+      data?: Uint8Array;
+      chainID: string;
+      version?: number;
+      options?: number;
+      guardian?: string;
+    }) {
+      this.nonce = nonce || 0;
+      this.value = value || new BigNumber(0);
+      this.sender = sender;
+      this.receiver = receiver;
+      this.senderUsername = senderUsername || "";
+      this.receiverUsername = receiverUsername || "";
+      this.gasPrice = gasPrice || new BigNumber(TRANSACTION_MIN_GAS_PRICE);
+      this.gasLimit = gasLimit;
+      this.data = data || new Uint8Array();
+      this.chainID = chainID;
+      this.version = version || TRANSACTION_OPTIONS_DEFAULT;
+      this.options = options || TRANSACTION_OPTIONS_DEFAULT;
+      this.guardian = guardian || "";
+  
+      this.signature = new Uint8Array();
+      this.guardianSignature = new Uint8Array();
+    }
+}
+
+/**
+ * An utilitary class meant to work together with the {@link TransactionNext} class.
+ */
+export class TransactionComputer {
+  constructor() {}
+
+  computeTransactionFee(transaction: ITransactionNext, networkConfig: INetworkConfig): BigNumber{
+    let moveBalanceGas = new BigNumber(
+      networkConfig.MinGasLimit + transaction.data.length * networkConfig.GasPerDataByte);
+    if (moveBalanceGas > transaction.gasLimit) {
+      throw new errors.ErrNotEnoughGas(parseInt(transaction.gasLimit.toString(), 10));
+    }
+
+    let gasPrice = new BigNumber(transaction.gasPrice);
+    let feeForMove = moveBalanceGas.multipliedBy(gasPrice);
+    if (moveBalanceGas === transaction.gasLimit) {
+      return feeForMove;
+    }
+
+    let diff = new BigNumber(transaction.gasLimit).minus(moveBalanceGas);
+    let modifiedGasPrice = gasPrice.multipliedBy(
+      new BigNumber(networkConfig.GasPriceModifier)
+    );
+    let processingFee = diff.multipliedBy(modifiedGasPrice);
+
+    return feeForMove.plus(processingFee);
+  }
+
+  computeBytesForSigning(transaction: ITransactionNext): Uint8Array{
+    let plainTransaction = this.toPlainObject(transaction);
+
+    if (plainTransaction.signature) {
+      delete plainTransaction.signature;
+    }
+
+    if (plainTransaction.guardianSignature) {
+      delete plainTransaction.guardianSignature;
+    }
+
+    if (!plainTransaction.guardian) {
+      delete plainTransaction.guardian
```

Comment:
If the `guardian` field is missing, why do you need to try to delete it from `plainTransaction.guardian` ?

Reply:
Because if it's missing it shouldn't be included in the transaction signature. This is how it was also done before. For reference check [the previous implementation](https://github.com/multiversx/mx-sdk-js-core/blob/main/src/transaction.ts#L286).